### ISSUE TITLE
Rule Proposal: `vue/no-deprecated-delete-set`

### ISSUE THREAD

**Issue by alexanderturinske** [vuejs/eslint-plugin-vue#2526]
**Please describe what the rule should do:**

Since `$delete` and `$set` APIs are [removed in Vue 3](https://v3-migration.vuejs.org/breaking-changes/#removed-apis); it would be helpful to add a warning to help people transition to Vue 3 by discouraging deprecated APIs

`Global functions set and delete, and the instance methods $set and $delete. They are no longer required with proxy-based change detection.`

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[x] Other (please specify: Vue 3 transition)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- BAD -->
this.$set(...)
Vue.set(...)
this.$delete(...)
Vue.delete(...)
```

```vue
<!-- BAD -->
Vue.set(col, 'value', usage[col.id]);

<!-- GOOD -->
Object.assign(col, { value: usage[col.id] });
```

```vue
<!-- BAD -->
this.$set(targetObj, lastKey, value);

<!-- GOOD -->
targetObj[lastKey] = value;
```

```vue
<!-- BAD -->
Vue.delete(targetObj, lastKey);
this.$delete(targetObj, lastKey);

<!-- GOOD -->
delete targetObj[lastKey];
```

**Additional context**

- [Vue 3 docs](https://v3-migration.vuejs.org/breaking-changes/#removed-apis)
- [Custom ESLint rules for preventing $delete in Vue files](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/162079)
- [GitLab: Remove Vue.set occurrences epic](https://gitlab.com/gitlab-org/gitlab/-/issues/452467)
- [GitLab: Remove Vue.delete occurrences epic](https://gitlab.com/gitlab-org/gitlab/-/issues/451128)

**Comment by FloEdelmann** [comment]
Additionally, these cases should be reported:

```ts
import { set, del } from 'vue'

set(obj, key, val)
del(obj, key)
```

```ts
const { set, del } = require('vue')

set(obj, key, val)
del(obj, key)
```
